src/main/java/com/abcmart/shoestore/application/OrderService.java

```diff
@@ -0,0 +1,127 @@
+        OrderEntity orderEntity = OrderEntity.create(detailEntities, orderPayment);
+        OrderEntity savedEntity = orderRepository.save(orderEntity);
+
+        return Order.from(savedEntity);
+    }
+
+    private BigDecimal calculateTotalPrice(Map<Long, ShoeEntity> shoeEntityMap,
+        List<CreateOrderRequest.CreateOrderDetailRequest> orderDetails) {
+
+        BigDecimal total = BigDecimal.ZERO;
+        for (CreateOrderRequest.CreateOrderDetailRequest orderDetail : orderDetails) {
+
+            Long shoeCode = orderDetail.getShoeCode();
+            ShoeEntity shoeEntity = shoeEntityMap.get(shoeCode);
+            if (Objects.isNull(shoeEntity)) {
+                continue;
+            }
+
+            BigDecimal shoesPrice = shoeEntity.getPrice()
+                .multiply(BigDecimal.valueOf(orderDetail.getCount()));
+            total = total.add(shoesPrice);
+        }
+
+        return total;
+    }
+
+    @Transactional
+    public Order cancelOrder(Long orderNo) {
+
+        OrderEntity targetOrderEntity = orderRepository.findByOrderNo(orderNo);
+        targetOrderEntity.totalCancel();
+        OrderEntity savedEntity = orderRepository.save(targetOrderEntity);
+
+        return Order.from(savedEntity);
+    }
+
+    @Transactional
+    public Order partialCancel(Long orderNo, Long shoeCode, Long removeCount) {
+
+        ShoeEntity shoeEntity = shoeRepository.findByShoeCode(shoeCode);
+
+        OrderEntity orderEntity = orderRepository.findByOrderNo(orderNo);
+        orderEntity.partialCancel(shoeEntity, removeCount);
+        OrderEntity savedEntity = orderRepository.save(orderEntity);
+
+        return Order.from(savedEntity);
+    }
+
+    @Transactional
+    public ShoeSaleCountResponse getShoeSaleCount() {
+
+        List<OrderDetailEntity> orderDetailEntityList = orderRepository.findAllNormalStatusOrderDetails();
+
+        List<Long> soldShoeCodes = orderDetailEntityList.stream()
+            .map(OrderDetailEntity::getShoeCode).toList();
+        Map<Long, ShoeEntity> soldShoeEntityMap = shoeRepository.findAllByShoeCodes(soldShoeCodes)
+            .stream()
+            .collect(Collectors.toMap(ShoeEntity::getShoeCode, Function.identity()));
+
+        HashMap<Long, SoldShoe> soldShoeHashMap = new HashMap<>();
+        orderDetailEntityList.forEach(detailEntity -> {
+            ShoeEntity shoeEntity = soldShoeEntityMap.get(detailEntity.getShoeCode());
+            if (Objects.isNull(shoeEntity)) {
+                return;
+            }
+
+            Shoe shoe = Shoe.from(shoeEntity);
+            Long saleCount = detailEntity.getCount();
+
+            if (soldShoeHashMap.containsKey(shoe.getShoeCode())) {
+                SoldShoe soldShoe = soldShoeHashMap.get(shoe.getShoeCode());
+                soldShoe.updateSaleCountAndTotalPrice(saleCount);
+                return;
+            }
+
+            soldShoeHashMap.put(shoe.getShoeCode(), SoldShoe.of(shoe, saleCount));
+        });
```

Comment:
```suggestion
        Map<Long, SoldShoe> soldShoeMap = orderDetailEntityList.stream()
    .filter(detail -> soldShoeEntityMap.containsKey(detail.getShoeCode()))
    .collect(Collectors.toMap(
        OrderDetailEntity::getShoeCode,
        detail -> {
            Shoe shoe = Shoe.from(soldShoeEntityMap.get(detail.getShoeCode()));
            return SoldShoe.of(shoe, detail.getCount());
        },
        (existing, added) -> {
            existing.updateSaleCountAndTotalPrice(added.getSaleCount());
            return existing;
        },
        HashMap::new
    ));
```
중간에 여러 조건으로 return 하는 게 조금 흐름이 흐려져서 
이렇게 할 수도 있을 것 같아요~

Reply:
오... 진짜 자바 잘쓰시네
감사합니다. 긍정적으로 검토해보겠습니다.

Reply:
resolved de5e19ab4c65d755916cd7ce110b633711452569